src/jwplayer/utils/useJwpAdsSetupComplete.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react';
+import { communicationService, ofType } from 'jwplayer/utils/communication';
+import { first } from 'rxjs/operators';
+import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
+
+export default function useJwpAdsSetupComplete(): boolean {
+	const [jwpAdsSetupComplete, setJwpAdsSetupComplete] = useState(false);
```

Comment:
I noticed that it is common in this repo to have a hook with state and `useEffect` that waits for some events and sets the state value. IMO it's not a good approach, especially when the hooks are used multiple times - we unnecessarily create duplicated states and duplicated event handlers. I think we should use react context instead or introduce some kind of state management library. Not necessarily something to do now but we should refactor it at some point.

Reply:
I've also observed that. Generally, the use of `useEffect` should be minimized since it's often unnecessary. However, I agree that discussing this falls beyond the scope of this PR, as you mentioned.

Reply:
Great feedback! I'll take a look at `PlayerContext` I saw in our code and see if this is something we could base on.